base/compiler/ssair/inlining.jl

```diff
@@ -994,6 +1017,22 @@ function inline_invoke!(ir::IRCode, idx::Int, sig::Signature, invoke_data::Invok
     return nothing
 end
 
+function narrow_yakc!(ir, stmt, calltype)
+    if isa(calltype, PartialYAKC)
+        lbt = argextype(stmt.args[3], ir, ir.sptypes)
+        lb, exact = instanceof_tfunc(lbt)
+        exact || return
+        # Narrow yakc type
+        if isa(calltype.ci, OptimizationState)
+            stmt.args[3] = stmt.args[4] = widenconst(tmerge(lb, calltype.ci.src.rettype))
+        elseif isa(calltype.ci, Method)
+            m = calltype.ci
+            stmt.args[5] = m
+            stmt.args[3] = stmt.args[4] = widenconst(tmerge(lb, m.unspecialized.cache.rettype))
```

Comment:
Is `m.unspecialized` guaranteed to be defined here? (Maybe I just need to keep reading)